### PR TITLE
TINKERPOP-2010 Generate JSDoc for gremlin-javascript

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2010

Docs generate with a standard build. Fastest way to see them is with `mvn process-resources -pl gremlin-javascript` which will generate them to `gremlin-javascript/src/main/javascript/gremlin-javascript/doc/`.

Updated `publish-docs.sh` to push jsdoc to `http://tinkerpop.apache.org/jsdocs/x.y.z/` on the website. Haven't actually executed that shell script yet.

VOTE +1

### CODE REVIEW

**Comment by dkuppitz** [comment]
Looks good to me.

VOTE +1

File: gremlin-javascript/glv/PackageJson.template [hunk 1]
```diff
@@ -33,7 +33,10 @@
   ],
   "license": "Apache-2.0",
   "dependencies": {
-    "ws": "^3.0.0"
+    "ws": "^3.0.0",
+    "grunt": "~1.0.2",
```
Comment (jorgebay) [inline gremlin-javascript/glv/PackageJson.template]: These 3 dependencies should be added under `devDependencies` instead

Comment (spmallette) [reply gremlin-javascript/glv/PackageJson.template]: ok - fixed

**Comment by jorgebay** [comment]
Awesome, thanks!
VOTE +1

**Pull request merged**